vortex-expr/src/binary.rs

```diff
@@ -82,6 +83,66 @@ pub(crate) mod proto {
     }
 }
 
+impl AnalysisExpr for BinaryExpr {
+    fn prune_expr(&self, catalog: &mut dyn StatsCatalog) -> Option<ExprRef> {
+        match self.operator {
+            Operator::Eq => {
+                let min_lhs = self.lhs.min(catalog);
+                let max_lhs = self.lhs.max(catalog);
+
+                let min_rhs = self.rhs.min(catalog);
+                let max_rhs = self.rhs.max(catalog);
+
+                let left = min_lhs.zip_with(max_rhs, gt);
+                let right = min_rhs.zip_with(max_lhs, gt);
+                left.into_iter().chain(right).reduce(or)
```

Comment:
Hmm. Perhaps I'm missing something subtle but I find the zip_withs and chains and reduces harder to read than ~~MONADIC DO NOTATION~~ question marks:

```suggestion
                let min_lhs = self.lhs.min(catalog)?;
                let max_lhs = self.lhs.max(catalog)?;

                let min_rhs = self.rhs.min(catalog)?;
                let max_rhs = self.rhs.max(catalog)?;

                or(gt(min_lhs, max_rhs), gt(min_rhs, max_lhs))
```

Reply:
The zip and reduce will find more possible exprs, (in theory), another valid pruning expr is `gt(min_lhs, max_rhs)` if the rhs cannot be created

Reply:
Ah right. Grr. Man that's pretty subtle. OK.

Reply:
If we have try2, it would be nicer